internal/cli/atlas_backups_restores_start.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/mongodb/mongocli/internal/json"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	automatedRestore = "AUTOMATED_RESTORE"
+	httpRestore      = "HTTP"
+)
+
+type atlasBackupsRestoresStartOpts struct {
+	*globalOpts
+	method               string
+	clusterName          string
+	clusterID            string
+	targetProjectID      string
+	targetClusterID      string
+	targetClusterName    string
+	checkpointID         string
+	oplogTs              string
+	oplogInc             int64
+	snapshotID           string
+	expirationHours      int64
+	expires              string
+	maxDownloads         int64
+	pointInTimeUTCMillis float64
+	store                store.ContinuousJobCreator
+}
+
+func (opts *atlasBackupsRestoresStartOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasBackupsRestoresStartOpts) Run() error {
+	request := opts.newContinuousJobRequest()
+
+	result, err := opts.store.CreateContinuousRestoreJob(opts.ProjectID(), opts.fromCluster(), request)
+
+	if err != nil {
+		return err
+	}
+
+	return json.PrettyPrint(result)
+}
+
+func (opts *atlasBackupsRestoresStartOpts) newContinuousJobRequest() *atlas.ContinuousJobRequest {
+	request := new(atlas.ContinuousJobRequest)
+	request.Delivery.MethodName = opts.method
+	request.SnapshotID = opts.snapshotID
+
+	if opts.isAutomatedRestore() {
+		request.Delivery.TargetGroupID = opts.targetProjectID
+		opts.setTargetCluster(request)
+
+		if opts.oplogTs != "" && opts.oplogInc != 0 {
+			request.OplogTs = opts.oplogTs
+			request.OplogInc = opts.oplogInc
+		}
+		if opts.pointInTimeUTCMillis != 0 {
+			request.PointInTimeUTCMillis = opts.pointInTimeUTCMillis
+		}
+	}
+
+	if opts.isHTTP() {
+		if opts.expires != "" {
+			request.Delivery.Expires = opts.expires
+		}
+		if opts.maxDownloads > 0 {
+			request.Delivery.MaxDownloads = opts.maxDownloads
+		}
+		if opts.expirationHours > 0 {
+			request.Delivery.ExpirationHours = opts.expirationHours
+		}
+	}
+	return request
+}
+
+func (opts *atlasBackupsRestoresStartOpts) fromCluster() string {
+	if opts.clusterName != "" {
+		return opts.clusterName
+	}
+	return opts.clusterID
+}
+
+func (opts *atlasBackupsRestoresStartOpts) setTargetCluster(out *atlas.ContinuousJobRequest) {
+	if opts.targetClusterID != "" {
+		out.Delivery.TargetClusterID = opts.targetClusterID
+	} else if opts.targetClusterName != "" {
+		out.Delivery.TargetClusterName = opts.targetClusterName
+	}
+}
+
+func (opts *atlasBackupsRestoresStartOpts) isAutomatedRestore() bool {
+	return opts.method == automatedRestore
+}
+
+func (opts *atlasBackupsRestoresStartOpts) isHTTP() bool {
+	return opts.method == httpRestore
+}
+
+func (opts *atlasBackupsRestoresStartOpts) validateParams() error {
+	if opts.clusterName == "" && opts.clusterID == "" {
```

Comment:
I am wondering if we should use a switch here

Reply:
this definitely needs more love, I'll see what I can do today with fresh eyes